src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java

```diff
@@ -543,16 +605,16 @@ public Object run() {
             }
         });
 
-        // clone a new stack not interfering with the current stack
-        AuthorizationStack newStack = RuntimeEnvironment.getInstance().getPluginStack().clone();
-
-        // increase the current plugin version tracked by the framework
-        increasePluginVersion();
+        // Clone a new stack not interfering with the current stack.
+        AuthorizationStack newStack = getStack().clone();
```

Comment:
Only thing here is that maybe I overlooked where will be the new stack comming from the configuration change?

Reply:
`RuntimeEnvironment#applyConfig()` now contains this:

```
         // set the new plugin directory and reload the authorization framework
         getAuthorizationFramework().setPluginDirectory(config.getPluginDirectory());
+        getAuthorizationFramework().setStack(config.getPluginStack());
         getAuthorizationFramework().reload();
```

Reply:
but now what happens if a request comes between these two lines
```java
         getAuthorizationFramework().setStack(config.getPluginStack());
         // <-- request for authorization
         getAuthorizationFramework().reload();
```

Reply:
good point. I guess delayed reconfiguration can be used to address that.